File: frontend/src/shared/models/post.ts
export type post = {
  id: string
  title: string
  body: string
  userId: string
  userName: string
  createdAt: string
  updatedAt: string
}

export const MOCK_POSTS: Array<post> = [
  {
    id: 'mock-id',
    title: 'モックタイトル',
    body: 'hoge',
    userId: 'mock-user-id1',
    userName: 'Funobu',
    createdAt: '2024-06-02T00:00:00+09:00',
    updatedAt: '2024-06-02T00:00:00+09:00'
  },
  {
    id: '1',
    title: 'モックタイトル1',
    body: 'こんにちは\nこんにちは',
    userId: 'mock-user-id1',
    userName: 'わだ',
    createdAt: '2024-06-02T00:00:00+09:00',
    updatedAt: '2024-06-02T00:00:00+09:00'
  },
  {
    id: '2',
    title: 'モックタイトル2',
    body: 'こんにちは\nこんにちは',
    userId: 'mock-user-id1',
    userName: 'わだ',
    createdAt: '2024-06-02T00:00:00+09:00',
    updatedAt: '2024-06-02T00:00:00+09:00'
  },
  {
    id: '3',
    title: 'モックタイトル3',
    body: 'こんにちは\nこんにちは',
    userId: 'mock-user-id1',
    userName: 'わだ',
    createdAt: '2024-06-02T00:00:00+09:00',
    updatedAt: '2024-06-02T00:00:00+09:00'
  },
  {
    id: '4',
    title: 'モックタイトル4',
    body: 'こんにちは\nこんにちは',
    userId: 'mock-user-id1',
    userName: 'わだ',
    createdAt: '2024-06-02T00:00:00+09:00',
    updatedAt: '2024-06-02T00:00:00+09:00'
  }
]

File: backend/domain/model/post.go
package model

import "time"

// Post は投稿 (body抜き)
// | 主に投稿一覧で使用するドメインモデルで、Bodyまで取得したい場合は PostDetail を使う
type Post struct {
	ID        int
	Title     string
	UserID    int
	UserName  string
	CreatedAt time.Time
	UpdatedAt time.Time
}

func (p Post) IsEmpty() bool {
	return p.ID == 0
}

// PostDetail は投稿の詳細
// | Bodyは長文のため、詳細の時にしか取得しないようにしている
type PostDetail struct {
	ID        int
	Title     string
	Body      string
	UserID    int
	UserName  string
	CreatedAt time.Time
	UpdatedAt time.Time
}

func (p PostDetail) IsEmpty() bool {
	return p.ID == 0
}
